jwst/tweakreg/tweakreg_step.py

```diff
@@ -200,12 +198,75 @@ def process(self, input):
+                # Raise Exception here to avoid rest of code in this try block
+                self.log.warning(msg)  
+            else:
+                # Set group_id to same value so all get fit as one observation
+                # The assigned value, 987654, has been hard-coded to make it
+                # easy to recognize when alignment to GAIA was being performed
+                # as opposed to the group_id values used for relative alignment
+                # earlier in this step.
+                for imcat in imcats:
+                    imcat.meta['orig_group_id'] = imcat.meta['group_id']
+                    imcat.meta['group_id'] = 987654
+                    if 'REFERENCE' in imcat.meta['fit_info']['status']:
+                        del imcat.meta['fit_info']
+
+                # Perform fit
+                align_tab = align_wcs(imcats,
+                                      refcat=ref_cat,
+                                      enforce_user_order=False,
+                                      expand_refcat=False,
+                                      minobj=self.minobj,
+                                      match=tpmatch,
+                                      fitgeom=self.fitgeometry,
+                                      nclip=self.nclip,
+                                      sigma=(self.sigma, 'rmse')
+                                     )
+                # Reset group_id to original values
+                # Also, update/create the WCS .name attribute with information on this astrometric fit
+                for imcat in imcats:
+                    imcat.meta['group_id'] = imcat.meta['orig_group_id']
+
+
         for imcat in imcats:
             imcat.meta['image_model'].meta.cal_step.tweakreg = 'COMPLETE'
             # retrieve fit status and update wcs if fit is successful:
             fit_info = imcat.meta.get('fit_info')
-            if fit_info['status'] in 'SUCCESS':
+            if 'SUCCESS' in fit_info['status']:
                 imcat.meta['image_model'].meta.wcs = imcat.wcs
+                # NOTE: This .name attribute needs to be defined using a convention
+                #       agreed upon by the JWST Cal Working Group.
+                #       Current value is merely a place-holder based on HST conventions
+                #       This value should also be translated to the FITS WCSNAME keyword
+                #       IF that is what gets recorded in the archive for end-user searches
+                imcat.meta['image_model'].meta.wcs.name = "FIT-LVL3-{}".format(self.gaia_catalog)
+                
+                # Also update FITS representation 
+                gwcs_header = imcat.wcs.to_fits_sip(max_pix_error=0.1, 
+                                                max_inv_pix_error=0.1,
+                                                degree=3, 
+                                                npoints=128)
+                imcat.meta['image_model'].wcs = wcs.WCS(header=gwcs_header)
```

Comment:
I think this block is what is causing the problem in the existing regression tests.  If I delete this whole block, everything works fine.  Line 256 updates the `gwcs` object, and that's all that needs to be done.  We don't use or update any FITS WCS objects until we get rectified, resampled images, so we don't need to do that here.  And we are not currently not using the `gwcs.WCS.name` attribute, but we could in the future.  But any name including `FIT` and `GAIA` should be controlled by `self.align_to_gaia` of the step.

Reply:
This block of code deals with updating the input images's FITS WCS, something which apparently is not required.  It has been commented out in the latest commit, while the addition of the '.name' attribute to the WCS has been (rightfully) moved under the logic for successful GAIA fit only.  I would argue that this is required as the ONLY indication to the end-user (short of diving into the GWCS and hoping to recognize one of the elements there) that any additional alignment to GAIA has been performed on the input GWCS objects.  It is a key feature of the HST alignment strategy with INS and end-users already relying on that name.

Reply:
As of right now we aren't including a FITS SIP representation in the unrectified products, so you're right that this update is not needed right now. There's a ticket in the backlog to add the SIP, so this code code be reactivated whenever that gets implemented.